### PR TITLE
Update Caddy reverse proxy documentation

### PR DESCRIPTION
**Pull request by mholt**
Improve/simplify Caddy examples. Remove Caddy v1 (has long been EOL'ed)

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by clokep** [comment]
Removing this from the review queue for now. Let us know when you get a chance to finish the checklist above, in particular:

* Adding the newsfile (should be `changelog.d/13344.doc` with something describing the change).
* [Signing off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off) on the pull request -- this can be done by adding another commit or updating the description or leaving a new comment.

**Comment by mholt** [comment]
@clokep Thanks. Done!

**Comment by moritzdietz** [comment]
When checking the spec https://spec.matrix.org/latest/client-server-api/#web-browser-clients I was wondering if there is a need to add the other CORS headers back again. As of now it says they are "recommended".

File: changelog.d/13344.doc [hunk 1]
```diff
@@ -0,0 +1 @@
+Improve Caddy reverse proxy documentation
```
Comment (erikjohnston) [inline changelog.d/13344.doc:1]:
```suggestion
Improve Caddy reverse proxy documentation.
```

**Review by erikjohnston — APPROVED** [review]
Thanks! I'm going to trust you that the syntax is right :smile:

**Comment by mholt** [comment]
Hope so, I've done this a few times, ha :sweat_smile:

**Pull request merged**